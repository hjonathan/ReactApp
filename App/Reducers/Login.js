import {handleActions} from "redux-actions";
import actions from "../Actions";

const initialState = {
    isLoggedIn: false,
    isGoogleLoggedIn: false,
    credentials: null,
    error: null
};
export default handleActions(
    {
        /**
         * Reducer for "login/success" action
         * @param state
         * @param action
         */
        [actions.login.success]: (state, action) => ({
            ...state,
            credentials: action.payload.credentials,
            error: null,
            isLoggedIn: true
        }),
        /**
         * Reducer for "login/error" action
         * @param state
         * @param action
         */
        [actions.login.error]: (state, action) => ({
            ...state,
            error: action.payload.error
        }),
        /**
         * Reducer for "credentials/update" action
         * @param state
         * @param action
         */
        [actions.credentials.update]: (state, action) => ({
            ...state,
            credentials: action.payload,
            error: null
        }),
        /**
         * Reducer for "logout/success" action
         * @param state
         * @param action
         */
        [actions.logout.success]: (state, action) => ({
            ...state,
            credentials: null,
            error: null,
            isLoggedIn: false,
            isGoogleLoggedIn: false
        }),
        /**
         * Reducer for "googleLogin/success" action
         * @param state
         * @param action
         */
        [actions.googleLogin.success]: (state, action) => ({
            ...state,
            credentials: action.payload.credentials,
            error: null,
            isLoggedIn: true,
            isGoogleLoggedIn: true
        }),
        /**
         * Resets states
         * @param state
         * @param action
         * @returns {object}
         */
        [actions.oauth.reset]: () => (initialState)
    },
    initialState
);
