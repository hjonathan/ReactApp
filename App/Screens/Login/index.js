import {GoogleSignin} from "react-native-google-signin";
import {connect} from "react-redux";
import LoginScreen from "./login";
import actions from "../../Actions";

/**
 * Return state from Store to View Component
 * @param state
 */
const mapStateToProps = state => ({
    error: state.Oauth.error || null,
    isLoggedIn: state.Oauth.isLoggedIn
});

const mapDispatchToProps = dispatch => ({
    /**
     * Method to get the user oauth credentials from restler service.
     */
    onLogin: (username, password) => dispatch(actions.login.request({username, password})),
    /**
     * request user data
     */
    requestUserData: params => dispatch(actions.userData.request(params)),

    /**
     * Save the device token
     * @param {*} token generated by the device.
     */
    saveTokenStorage: (token) => {
        dispatch(actions.notification.token.save(token));
    },

    signInGoogle: (callback) => {
        GoogleSignin.signIn()
            .then((user) => {
                callback(null, user);
                dispatch(actions.googleLogin.request({
                    mail: user.email,
                    token: user.accessToken,
                    clientid: user.id
                }));
            })
            .catch((err) => {
                callback(err, null);
            })
            .done();
    },
    /**
     * Applies the necessary configuration to enable Google SignIn
     */
    setupGoogleSignIn: async (callback) => {
        try {
            await GoogleSignin.hasPlayServices({autoResolve: true});
            await GoogleSignin.configure({
                iosClientId: "592781414011-evmj9utjj34m3ltu9lumfndbme1c196p.apps.googleusercontent.com",
                webClientId: "592781414011-knieb4v9rnattp8k2ljqgdng5sdqvmk5.apps.googleusercontent.com",
                offlineAccess: false
            });

            const user = await GoogleSignin.currentUserAsync();
            callback(null, user);
        } catch (err) {
            callback(err);
        }
    },
    /**
     * Method to remove session from Google (SignOut Google).
     * @param {*} callback Confirm the removed session from Google.
     */
    signOutGoogle (callback) {
        GoogleSignin.revokeAccess().then(() => GoogleSignin.signOut()).then(() => {
            callback(null, null);
        }).done();
    }
});

export default connect(mapStateToProps, mapDispatchToProps)(LoginScreen);
