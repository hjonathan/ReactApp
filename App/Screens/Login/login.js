import React, {Component} from "react";
import {
    Image,
    Text,
    TextInput,
    View,
    ScrollView,
    Platform,
    I18nManager,
    PushNotificationIOS
} from "react-native";
import {NavigationActions} from "react-navigation";
import Orientation from "react-native-orientation";
import {Button} from "react-native-elements";
import PropTypes from "prop-types";
import Toast from "react-native-easy-toast";
import {ProgressDialog} from "react-native-simple-dialogs";
import PushNotification from "react-native-push-notification";
import Utils from "../../Utils";
import Actions from "../../Actions";
import I18n from "../../I18n";
import FlatButton from "../../Components/FlatButton";
import GoogleSignInButton from "../../Components/GoogleSignInButton";
import * as Res from "../../Assets/resources";
import Styles from "./styles";

/**
 * Class Login
 */
class Login extends Component {
    /**
     * Constructor Login
     * @param props
     */
    constructor (props) {
        super(props);
        this.state = {
            username: "",
            password: "",
            user: null,
            errorUserName: true,
            errorPassWord: true,
            loader: false,
            toastMessage: null,
            userDataRequested: false
        };
        Orientation.unlockAllOrientations();
    }

    /**
     * Initializes settings
     */
    componentWillMount () {
        this.initSettings();
    }

    /**
     * Init config signIn
     */
    componentDidMount () {
        Orientation.unlockAllOrientations();
        this.setupGoogleSignIn();
    }

    /**
     * Component will receive props
     */
    componentWillReceiveProps (nextProps) {
        if (nextProps.isLoggedIn || nextProps.error) {
            this.setState({loader: false});
        }
        this.setState({toastMessage: (nextProps.error && nextProps.error.error_description) || null});
        this.props.error = null;
    }

    /**
     * Component did update
     */
    componentDidUpdate () {
        if (this.state.toastMessage) {
            this.refs.toast.show(this.state.toastMessage);
        }
    }

    /**
     * Update user name
     * @param text
     * @returns {void|*}
     */
    onChangeUser = (text) => {
        this.setState({errorUserName: (text.length > 0), username: text, toastMessage: null});
    };

    /**
     * Update password
     * @param text
     * @returns {void|*}
     */
    onChangePassword = (text) => {
        this.setState({errorPassWord: (text.length > 0), password: text, toastMessage: null});
    };

    /**
     * Pass params
     */
    onLogin = () => {
        const name = this.state.username;
        const pass = this.state.password;
        if (name && pass) {
            this.setState({loader: true});
            this.props.onLogin(this.state.username, this.state.password);
            this.initNotification();
        }
        this.setState({
            errorUserName: name.length > 0,
            errorPassWord: pass.length > 0
        });
    };

    /**
     * Configuration google signIn
     */
    setupGoogleSignIn = () => this.props.setupGoogleSignIn((err, user) => {
        if (!err) {
            this.setState({user});
        }
    });

    /**
     * SignIn with google count
     */
    signInGoogle = () => this.props.signInGoogle((err, data) => {
        if (!err) {
            this.setState({loader: true, user: data});
        }
    });

    /**
     * SignOut google count
     */
    signOutGoogle = () => this.props.signOutGoogle((err, data) => {
        if (!err) {
            this.setState({user: data});
        }
    });

    /**
     * Gets the previous settings
     * @returns {Promise<void>}
     */
    async initSettings () {
        let defaultSettings = await Utils.multiGet(["@server", "@workspace", "@lang"]);
        this.props.navigation.dispatch(Actions.settings.updateSettings({
            url: defaultSettings["@server"],
            workspace: defaultSettings["@workspace"],
            lang: defaultSettings["@lang"]
        }));
    }

    /**
     * Initializes the notification variables
     */
    initNotification = () => {
        let that = this;
        PushNotification.configure({
            // (optional) Called when Token is generated (iOS and Android)
            onRegister (token) {
                that.props.saveTokenStorage(token);
            },

            // (required) Called when a remote or local notification is opened or received
            onNotification (notification) {
                // process the notification
                // TODO build open case
                // required on iOS only (see fetchCompletionHandler docs: https://facebook.github.io/react-native/docs/pushnotificationios.html)
                if (Platform.OS === "ios") {
                    // eslint-disable-next-line no-undef
                    notification.finish(PushNotificationIOS.FetchResult.NoData);
                }
            },

            // ANDROID ONLY: GCM Sender ID (optional - not required for local notifications, but is need to receive remote push notifications)
            senderID: "592781414011",

            // IOS ONLY (optional): default: all - Permissions to register.
            permissions: {
                alert: true,
                badge: true,
                sound: true
            },

            // Should the initial notification be popped automatically
            // default: true
            popInitialNotification: true,

            /**
             * (optional) default: true
             * - Specified if permissions (ios) and token (android and ios) will requested or not,
             * - if not, you must call PushNotificationsHandler.requestPermissions() later
             */
            requestPermissions: true
        });
        return PushNotification;
    };

    /**
     * Render error for User Name field.
     */
    _renderErrorUser () {
        if (!this.state.errorUserName) {
            return (
                this._errorViewForm(I18n.t("nullUser"))
            );
        }
        return null;
    }

    /**
     * Render error icon for Password field.
     */
    _renderErrorPass () {
        if (!this.state.errorPassWord) {
            return (
                this._errorViewForm(I18n.t("nullPass"))
            );
        }
        return null;
    }

    /**
     * Redirects to settings screen
     */
    goToSettings = () => {
        const routeToScreen = NavigationActions.navigate({
            routeName: "Settings",
            params: {
                logged: false
            }
        });
        this.props.navigation.dispatch(routeToScreen);
    };

    /**
     * Returns an error view for invalid values.
     * @param {string} message
     */
    _errorViewForm (message) {
        return (
            <View style={Styles.viewContainerError}>
                <Image
                    source={Res.images.errorIcon}
                    style={Styles.iconErrorField}
                />
                <Image
                    source={Res.images.errorArrowIcon}
                    style={Styles.arrowImageError}
                />
                <Text
                    style={Styles.textError}
                    adjustFontSizeToFit
                >{message}
                </Text>
            </View>
        );
    }

    /**
     * Renders the logo header
     * @returns {*}
     */
    renderLogo () {
        return (
            <View style={Styles.logoContainer}>
                <Image
                    source={Res.images.logo}
                    style={Styles.logo}
                />
            </View>
        );
    }

    /**
     * Renders divider line
     * @returns {*}
     */
    renderDivider () {
        return (
            <View style={{
                justifyContent: "center",
                flexDirection: "row",
                marginBottom: 15,
                marginTop: 15
            }}
            >
                <View style={{height: 1, backgroundColor: "#fff", width: "36%"}} />
                <View style={{width: 40, alignItems: "center"}}>
                    <Text style={{margin: -10, color: "#fff"}}>{I18n.t("login_or_label")}</Text>
                </View>
                <View style={{height: 1, backgroundColor: "#fff", width: "36%"}} />
            </View>
        );
    }

    /**
     * Renders login form
     * @returns {*}
     */
    renderForm () {
        const alignRTL = I18nManager.isRTL ? "right" : "left";
        return (
            <View>
                <View style={[Styles.subContainer, {marginTop: 20, zIndex: 0}]}>
                    <View style={Styles.textSectionStyle}>
                        <TextInput
                            style={[Styles.textBox, this.state.errorUserName && Styles.textBoxErr, {textAlign: alignRTL}]}
                            onChangeText={this.onChangeUser}
                            placeholder={I18n.t("login_user")}
                            underlineColorAndroid="transparent"
                        />
                    </View>
                    <View style={Styles.textSectionStyle}>
                        <TextInput
                            style={[Styles.textBox, this.state.errorPassWord && Styles.textBoxErr, {textAlign: alignRTL}]}
                            secureTextEntry
                            onChangeText={this.onChangePassword}
                            placeholder={I18n.t("login_password")}
                            underlineColorAndroid="transparent"
                        />
                    </View>
                </View>
                <View style={Styles.fieldContainer}>
                    <Button
                        raised
                        backgroundColor={Res.colors.successColor}
                        buttonStyle={{height: 38}}
                        borderRadius={2}
                        title={I18n.t("login_btn")}
                        fontSize={14}
                        fontWeight="bold"
                        onPress={() => this.onLogin()}
                    />
                </View>
                <View style={Styles.fieldContainer}>
                    <GoogleSignInButton
                        params={{
                            backgroundColor: "#4285F4",
                            color: Res.colors.white
                        }}
                        onPress={this.signInGoogle}
                        title={I18n.t("login_google_btn")}
                        icon={Res.images.googleSignInIcon}
                    />
                </View>
                {this.renderDivider()}
                <View style={{
                    justifyContent: "flex-end",
                    alignItems: "flex-end",
                    alignSelf: "flex-end",
                    position: "absolute"
                }}
                >{this._renderErrorUser()}
                </View>
                <View style={{
                    justifyContent: "flex-end",
                    alignItems: "flex-end",
                    alignSelf: "flex-end",
                    position: "absolute",
                    marginTop: 46
                }}
                >{this._renderErrorPass()}
                </View>
            </View>
        );
    }

    /**
     * Render
     * @returns {*}
     */
    render () {
        return (
            <View style={{flex: 1}}>
                <ScrollView
                    contentContainerStyle={{flexGrow: 1}}
                    keyboardShouldPersistTaps="always"
                    keyboardDismissMode="on-drag"
                >
                    <View style={{flex: 1}}>
                        <View style={Styles.mainContainer}>
                            {this.renderLogo()}
                            {this.renderForm()}
                            <View style={Styles.fieldContainer}>
                                <FlatButton
                                    params={{radius: 5, border: 1, color: Res.colors.white}}
                                    onPress={() => {
                                        this.goToSettings();
                                    }}
                                    title={I18n.t("menu_drawer_settings_item_settings")}
                                />
                            </View>
                            <ProgressDialog
                                visible={this.state.loader}
                                title={I18n.t("login_dialog_loading_title")}
                                message={I18n.t("login_dialog_loading_description")}
                            />
                            <Toast
                                ref="toast"
                                style={{margin: 20, backgroundColor: Res.colors.lightColor}}
                                textStyle={{color: "white", textAlign: "center"}}
                            />
                        </View>
                    </View>
                </ScrollView>
            </View>
        );
    }
}

Login.propTypes = {
    onLogin: PropTypes.func.isRequired,
    signInGoogle: PropTypes.func.isRequired,
    signOutGoogle: PropTypes.func.isRequired,
    setupGoogleSignIn: PropTypes.func.isRequired
};

export default Login;
