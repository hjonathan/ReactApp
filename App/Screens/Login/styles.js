import {StyleSheet} from "react-native";
import * as Res from "../../Assets/resources";

export default StyleSheet.create({
    mainContainer: {
        flex: 1,
        flexDirection: "column",
        alignItems: "center",
        justifyContent: "space-around",
        paddingHorizontal: 60,
        paddingVertical: 16,
        backgroundColor: Res.colors.mainColor
    },
    subContainer: {
        marginBottom: 6,
        marginTop: 6
    },
    logoContainer: {
        marginBottom: 16,
        justifyContent: "center",
        width: 220,
        height: 40
    },
    logo: {
        width: "100%",
        resizeMode: "contain"
    },
    fieldContainer: {
        alignContent: "center",
        marginBottom: 3,
        paddingTop: 10,
        zIndex: 0
    },
    textBox: {
        backgroundColor: "white",
        height: 40,
        borderWidth: 1.5,
        borderColor: "white",
        minWidth: 100
    },
    textBoxErr: {
        borderColor: "transparent"
    },
    textSectionStyle: {
        backgroundColor: "#fff",
        marginTop: 3,
        marginHorizontal: 14,
        height: 39
    },
    accept: {
        backgroundColor: Res.colors.successColor
    },
    icon: {
        color: "#fff",
        fontSize: 18,
        paddingLeft: 12,
        paddingRight: 12
    },
    container: {
        paddingBottom: 10,
        paddingTop: 10
    },
    googleSignInButton: {
        borderRadius: 2,
        height: 40,
        marginHorizontal: 14,
        justifyContent: "center"
    },
    iconErrorField: {
        margin: 5,
        marginTop: 2,
        height: 20,
        width: 20,
        resizeMode: "stretch",
        justifyContent: "flex-end"
    },
    textError: {
        marginTop: -3,
        backgroundColor: "black",
        color: "white",
        borderTopWidth: 2,
        borderTopColor: "red",
        paddingLeft: 3,
        paddingRight: 3,
        fontSize: 12,
        opacity: 0.8
    },
    arrowImageError: {
        width: 12,
        height: 12,
        marginTop: -9,
        marginRight: 7,
        resizeMode: "stretch"
    },
    viewContainerError: {
        flexDirection: "column",
        justifyContent: "flex-end",
        alignItems: "flex-end",
        marginTop: 31,
        marginHorizontal: 14
    }
});
