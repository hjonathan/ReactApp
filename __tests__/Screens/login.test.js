import React from 'react'
import {shallow} from 'enzyme';
import ConnectedLogin from '../../App/Screens/Login';
import {Login} from '../../App/Screens/Login/login';
import LoginScreen from '../../App/Screens/Login/login';
import configureStore from 'redux-mock-store'
jest.mock('react-native-google-signin', () => require.requireActual('../../__mocks__/react-native-google-signin').default);
jest.mock('react-native-orientation', () => require.requireActual('../../__mocks__/react-native-orientation').default);
import Orientation from 'react-native-orientation';
jest.mock('react-native-push-notification', () => require.requireActual('../../__mocks__/react-native-push-notification').default);
import PushNotification from 'react-native-push-notification';

import {createStore} from 'redux'

describe('Screen >>> Login', () => {
    const initialState = {
            Oauth: {
                error: {
                    error: "error",
                    error_description: "error_description"
                }
            },
            username: "",
            password: "",
            user: null,
            errorUserName: true,
            errorPassWord: true,
            loader: false,
            toastMessage: null
        },
        middlewares = [],
        mockStore = configureStore(middlewares),
        navigation = {
            dispatch: jest.fn(),
        };
    let store, container, viewLogin;
    const onLogin = () => {
        },
        signInGoogle = () => {
        },
        signOutGoogle = () => {
        },
        setupGoogleSignIn = () => {
        },
        _errorViewForm = () => {
        };

    beforeEach(() => {
        store = mockStore(initialState)
        container = shallow(<ConnectedLogin store={store}/>)
        viewLogin = shallow(<LoginScreen navigation={navigation} Orientation={Orientation} onLogin={onLogin} signInGoogle={signInGoogle}
                                         signOutGoogle={signOutGoogle} setupGoogleSignIn={setupGoogleSignIn}
                                         _errorViewForm={_errorViewForm} store={store} />)
    });

    it('Render the connected(SMART) component', () => {
        expect(container.length).toEqual(1)
    });

    it('mapStateToProps', () => {
        expect(container.props('children').onLogin()).toMatchSnapshot();
        expect(container.props('children').requestUserData()).toMatchSnapshot();
        expect(container.props('children').saveTokenStorage()).toMatchSnapshot();
        expect(container.props('children').signInGoogle()).toMatchSnapshot();
        expect(container.props('children').setupGoogleSignIn()).toMatchSnapshot();
        expect(container.props('children').signOutGoogle()).toMatchSnapshot();
    });

    it('Check Prop matches with initialState', () => {
        expect(container.prop('error')).toEqual({
            error: "error",
            error_description: "error_description"
        });
    });

    it('Props coverage componentWillReceiveProps', () => {
        viewLogin.instance().componentWillReceiveProps({
            error: "error",
            error_description: "error_description",
            isLoggedIn: true
        });
        expect(viewLogin.instance().state.errorPassWord).toEqual(true);
    });

    it('Props coverage componentDidUpdate', () => {
        viewLogin.instance().refs = {
            toast: {
                show: (text) => {}
            }
        };
        viewLogin.setState({toastMessage: "TestToast"});
        viewLogin.instance().componentDidUpdate();
        expect(viewLogin.instance().state.toastMessage).toEqual("TestToast");
    });

    it("Should call onChangeUser method", () => {
        expect(viewLogin.instance().onChangeUser("testUser")).toBeUndefined();
        expect(viewLogin.state().errorUserName).toEqual(true);
        expect(viewLogin.state().username).toEqual("testUser");
    });

    it("Should call onChangePassword method", () => {
        expect(viewLogin.instance().onChangePassword("testPass")).toBeUndefined();
        expect(viewLogin.state().errorPassWord).toEqual(true);
        expect(viewLogin.state().password).toEqual("testPass");
    });

    it("Should call onLogin method", () => {
        viewLogin.setState({username: "TestUser", password: "TestPass"});
        expect(viewLogin.instance().onLogin()).toBeUndefined();
        expect(viewLogin.state().loader).toEqual(true);
    });

    it("Should call setupGoogleSignIn method", () => {
        expect(viewLogin.instance().setupGoogleSignIn()).toBeUndefined();
    });

    it("Should call signInGoogle method", () => {
        expect(viewLogin.instance().signInGoogle()).toBeUndefined();
    });

    it("Should call signOutGoogle method", () => {
        expect(viewLogin.instance().signOutGoogle()).toBeUndefined();
    });

    it('Should call  initNotification method', () => {
        expect(viewLogin.instance().initNotification()).toEqual(PushNotification);
    });

    it("Should call _renderErrorUser method", () => {
        expect(viewLogin.instance()._renderErrorUser()).toBeNull();
        viewLogin.setState({errorUserName: null});
        expect(viewLogin.instance()._renderErrorUser()).toMatchSnapshot();
    });

    it("Should call _renderErrorPass method", () => {
        expect(viewLogin.instance()._renderErrorPass()).toBeNull();
        viewLogin.setState({errorPassWord: null});
        expect(viewLogin.instance()._renderErrorPass()).toMatchSnapshot();
    });

    it("Should call goToSettings method", () => {
        expect(viewLogin.instance().goToSettings()).toBeUndefined();
    });

    it('Should call _errorViewForm method', () => {
        expect(viewLogin.instance()._errorViewForm).toMatchSnapshot();
    });

    it('Should call renderLogo method', () => {
        expect(viewLogin.instance().renderLogo).toMatchSnapshot();
    });

    it('Should call renderDivider method', () => {
        expect(viewLogin.instance().renderDivider).toMatchSnapshot();
    });

    it('Should call renderForm method', () => {
        expect(viewLogin.instance().renderForm).toMatchSnapshot();
    });
});
