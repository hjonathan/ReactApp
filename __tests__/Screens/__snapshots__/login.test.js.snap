// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Screen >>> Login Should call _errorViewForm method 1`] = `[Function]`;

exports[`Screen >>> Login Should call _renderErrorPass method 1`] = `
<Component
  style={
    Object {
      "alignItems": "flex-end",
      "flexDirection": "column",
      "justifyContent": "flex-end",
      "marginHorizontal": 14,
      "marginTop": 31,
    }
  }
>
  <Image
    source={
      Object {
        "testUri": "../../../App/Assets/icons/error.png",
      }
    }
    style={
      Object {
        "height": 20,
        "justifyContent": "flex-end",
        "margin": 5,
        "marginTop": 2,
        "resizeMode": "stretch",
        "width": 20,
      }
    }
  />
  <Image
    source={
      Object {
        "testUri": "../../../App/Assets/icons/redArrowError.png",
      }
    }
    style={
      Object {
        "height": 12,
        "marginRight": 7,
        "marginTop": -9,
        "resizeMode": "stretch",
        "width": 12,
      }
    }
  />
  <Text
    accessible={true}
    adjustFontSizeToFit={true}
    allowFontScaling={true}
    ellipsizeMode="tail"
    style={
      Object {
        "backgroundColor": "black",
        "borderTopColor": "red",
        "borderTopWidth": 2,
        "color": "white",
        "fontSize": 12,
        "marginTop": -3,
        "opacity": 0.8,
        "paddingLeft": 3,
        "paddingRight": 3,
      }
    }
  >
    nullPass
  </Text>
</Component>
`;

exports[`Screen >>> Login Should call _renderErrorUser method 1`] = `
<Component
  style={
    Object {
      "alignItems": "flex-end",
      "flexDirection": "column",
      "justifyContent": "flex-end",
      "marginHorizontal": 14,
      "marginTop": 31,
    }
  }
>
  <Image
    source={
      Object {
        "testUri": "../../../App/Assets/icons/error.png",
      }
    }
    style={
      Object {
        "height": 20,
        "justifyContent": "flex-end",
        "margin": 5,
        "marginTop": 2,
        "resizeMode": "stretch",
        "width": 20,
      }
    }
  />
  <Image
    source={
      Object {
        "testUri": "../../../App/Assets/icons/redArrowError.png",
      }
    }
    style={
      Object {
        "height": 12,
        "marginRight": 7,
        "marginTop": -9,
        "resizeMode": "stretch",
        "width": 12,
      }
    }
  />
  <Text
    accessible={true}
    adjustFontSizeToFit={true}
    allowFontScaling={true}
    ellipsizeMode="tail"
    style={
      Object {
        "backgroundColor": "black",
        "borderTopColor": "red",
        "borderTopWidth": 2,
        "color": "white",
        "fontSize": 12,
        "marginTop": -3,
        "opacity": 0.8,
        "paddingLeft": 3,
        "paddingRight": 3,
      }
    }
  >
    nullUser
  </Text>
</Component>
`;

exports[`Screen >>> Login Should call renderDivider method 1`] = `[Function]`;

exports[`Screen >>> Login Should call renderForm method 1`] = `[Function]`;

exports[`Screen >>> Login Should call renderLogo method 1`] = `[Function]`;

exports[`Screen >>> Login mapStateToProps 1`] = `
Object {
  "payload": Object {
    "password": undefined,
    "username": undefined,
  },
  "type": "login/request",
}
`;

exports[`Screen >>> Login mapStateToProps 2`] = `
Object {
  "type": "userData/request",
}
`;

exports[`Screen >>> Login mapStateToProps 3`] = `undefined`;

exports[`Screen >>> Login mapStateToProps 4`] = `undefined`;

exports[`Screen >>> Login mapStateToProps 5`] = `
Promise {
  "_40": 0,
  "_55": null,
  "_65": 0,
  "_72": null,
}
`;

exports[`Screen >>> Login mapStateToProps 6`] = `undefined`;
